src/Package/Impl/DataInspect/DataImport/ImportDataWindow.xaml.cs

```diff
@@ -0,0 +1,309 @@
+            if (encodingModel != null && model.Value != null) {
+                parameter.AppendFormat(" ,encoding={0}", model.Value.ToRStringLiteral());
+            }
+
+            parameter.AppendFormat(" ,sep={0}", ComboBoxSelectedRStringLiteral(SeparatorComboBox));
+            parameter.AppendFormat(" ,dec={0}", ComboBoxSelectedRStringLiteral(DecimalComboBox));
+            parameter.AppendFormat(" ,quote={0}", ComboBoxSelectedRStringLiteral(QuoteComboBox));
+            parameter.AppendFormat(" ,comment.char={0}", ComboBoxSelectedRStringLiteral(CommentComboBox));
+
+            if (!string.IsNullOrEmpty(NAStringTextBox.Text)) {
+                parameter.Append(" ,na.strings=");
+                AppendString(parameter, NAStringTextBox.Text.ToRStringLiteral());
+            }
+
+            if (preview) {
+                parameter.Append(" ,nrows=20");
+                return string.Format(CultureInfo.InvariantCulture, "read.csv({0})", parameter);
+            } else {
+                return string.Format(CultureInfo.InvariantCulture, "`{0}` <- read.csv({1})", VariableNameBox.Text, parameter);
+            }
+        }
+
+        private string ComboBoxSelectedRStringLiteral(ComboBox comboBox) {
+            return ((ComboBoxItemModel)comboBox.SelectedItem).Value.ToRStringLiteral();
+        }
+
+        private void AppendString(StringBuilder builder, string value) {
+            value = value.Replace("\"", "\\\"");
+            builder.AppendFormat("\"{0}\"", value);
+        }
+
+        private void FileOpenButton_Click(object sender, RoutedEventArgs e) {
+            IVsUIShell uiShell = VsAppShell.Current.GetGlobalService<IVsUIShell>(typeof(SVsUIShell));
+            IntPtr dialogOwner;
+            uiShell.GetDialogOwnerHwnd(out dialogOwner);
+
+            FilePathBox.Text = VsAppShell.Current.BrowseForFileOpen(dialogOwner, "CSV file|*.csv");
+            if (!string.IsNullOrEmpty(FilePathBox.Text)) {
+                FilePathBox.CaretIndex = FilePathBox.Text.Length;
+                FilePathBox.ScrollToEnd();
+
+                var variableName = Path.GetFileNameWithoutExtension(FilePathBox.Text);
+                VariableNameBox.Text = variableName;
+
+                string text = ReadFile(FilePathBox.Text);
+                InputFilePreview.Text = text;
+
+                PreviewAsync().DoNotWait();
+            }
+        }
+
+        private void RunButton_Click(object sender, RoutedEventArgs e) {
+            RunButton.IsEnabled = false;
+            CancelButton.IsEnabled = false;
+
+            var expression = BuildCommandLine(false);
+            if (expression != null) {
+                RunAsync(expression).DoNotWait();
+            }
+        }
+
+        private void CancelButton_Click(object sender, RoutedEventArgs e) {
+            base.Close();
+        }
+
+        private async Task PreviewAsync() {
+            var expression = BuildCommandLine(true);
```

Comment:
This code is invoked on UI thread, so you don't have to call `DispatchOnUIThread`